### ISSUE TITLE
Checking kwarg names by identity isn't always safe

### ISSUE THREAD

**Issue by Dr-Emann** [ariebovenberg/whenever#149]
Version: `whenever==0.6.2`

Example code which fails unexpectedly:

```python
import whenever
import whenever._pywhenever
import json

t2 = whenever._pywhenever.Time()
t1 = whenever.Time()

shift_kwargs = json.loads("""{"hour": 3}""")

assert '03:00:00' == str(t2.replace(**shift_kwargs))
assert '03:00:00' == str(t1.replace(**shift_kwargs))
# ^- TypeError: replace() got an unexpected keyword argument: 'hour'
```

This is caused by comparing kwargs by pointer identity, e.g. here: https://github.com/ariebovenberg/whenever/blob/6b06aa9172c313c6840118c263940a30872e0721/src/time.rs#L425-L426

**Comment by ariebovenberg** [comment]
Nice catch! I was under the impression that the automatic interning of dict keys made this operation safe. What's probably going on here is that `json` (an extension module itself) bypasses the automatic interning of dictionary keys 🤔.

Note you may want to post your issue to orjson's issue tracker as well, since that's where I got this pattern ([code link](https://github.com/ijl/orjson/blob/4c05bfafdd2613e20b9be994ec3eaa846700a568/src/lib.rs#L343))

**Comment by Dr-Emann** [comment]
There's nothing special about `json` or dict keys: literals (including some simple constant folding) are always interned, but strings built any other way won't be equal by identity. e.g. this also fails in the same way:

```python
# python constant-folds "literal" + "literal", so we need to be just a little fancier
hour_str = "ho" + (lambda: "ur")()
shift_kwargs = { hour_str: 3 }
t1.replace(**shift_kwargs)
# ^- TypeError: replace() got an unexpected keyword argument: 'hour'
```

**Comment by ariebovenberg** [comment]
Interesting! I had based my reasoning off of [this line](https://github.com/python/cpython/blob/15d48aea02099ffc5bdc5511cc53ced460cb31b9/Objects/dictobject.c#L4890) in `PyDict_SetItemString`.

In any case, I'll make the adjustment.

**Comment by ariebovenberg** [comment]
@Dr-Emann The fix is out in version 0.6.3. I'm happy to have found a way to fix this while keeping the performance benefits from interning. I also got around to streamlining the kwarg handling so that on the whole things are now faster 🚀. 

Thanks again for posting and let me know if you spot anything else 🔍
